pkg/asset/manifests/dns.go

```diff
@@ -96,6 +98,14 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 		config.Spec.PrivateZone = &configv1.DNSZone{
 			ID: dnsConfig.GetDNSZoneID(clusterID.InfraID+"-rg", installConfig.Config.ClusterDomain()),
 		}
+	case gcptypes.Name:
+		zone, err := icgcp.GetPublicZone(installConfig.Config.BaseDomain, installConfig.Config.Platform.GCP.ProjectID)
+		if err != nil {
+			return errors.Wrapf(err, "getting public zone for %q", installConfig.Config.BaseDomain)
+		}
+		//TODO determine if zone.Name is the correct ID for this config
+		config.Spec.PublicZone = &configv1.DNSZone{ID: fmt.Sprint(zone.Id)}
+		//TODO: Determine what will be the name/id of created private zone
```

Comment:
The private zone is created by the installer.. So we can predict the name.
So on azure we know the ID https://github.com/openshift/installer/blob/45f81e1d950cbc80e46e0b230e11b0032de8c868/pkg/asset/manifests/dns.go#L96-L98

on AWS we know the tags to get the *single* zone. https://github.com/openshift/installer/blob/45f81e1d950cbc80e46e0b230e11b0032de8c868/pkg/asset/manifests/dns.go#L82-L85

based on https://cloud.google.com/dns/docs/reference/v1/managedZones/get  and https://github.com/openshift/api/blob/de5ca909c7322bb8d06fa5a9e5604491b373da52/config/v1/types_dns.go#L54
the ID gor GCP will be the name of the DNS zone, which will probably be `<infra-id>-private-zone`